processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/PartitionContexts.java

```diff
@@ -214,7 +233,9 @@ public List<TopicPartition> partitionsNeedsPause() {
         return contexts.values().stream()
                        .filter(c -> !c.revoking())
                        .filter(c -> !c.paused())
-                       .filter(c -> pausingAll || shouldPartitionPaused(c.pendingTasksCount()))
+                       .filter(c -> pausingAll
+                                    || reloadStates.get(c.topicPartition())
```

Comment:
Additionally, we also need to modify `partitionNeedsResume` to avoid returning partitions that reload is requested, because otherwise, partitions will be resumed immediately so new tasks will be kept feeding so pendingTasks never becomes 0.